src/commonMain/kotlin/fr/acinq/lightning/blockchain/electrum/ElectrumClient.kt

```diff
@@ -25,31 +25,30 @@ import kotlin.time.Duration.Companion.seconds
 
 
 /** Commands */
-internal sealed interface ElectrumClientCommand {
+sealed interface ElectrumClientCommand {
     object Connected : ElectrumClientCommand
     object Disconnected : ElectrumClientCommand
-    data class ReceivedResponse(val response: Either<ElectrumResponse, JsonRPCResponse>) : ElectrumClientCommand
-    data class SendElectrumApiCall(val electrumRequest: ElectrumRequest) : ElectrumClientCommand
-    object AskForHeader : ElectrumClientCommand
+    data class AskForHeader(val id: UUID) : ElectrumClientCommand
+    data class ReceivedElectrumResponse(val response: Either<ElectrumResponse, JsonRPCResponse>) : ElectrumClientCommand
+    data class SendElectrumRequest(val id: UUID, val electrumRequest: ElectrumRequest) : ElectrumClientCommand
 }
 
 /** Actions */
-internal sealed interface ElectrumClientAction {
+sealed interface ElectrumClientAction {
+    data class SendHeader(val ids: Set<UUID>, val height: Int, val blockHeader: BlockHeader) : ElectrumClientAction
     data class SendRequest(val request: String) : ElectrumClientAction
-    data class SendHeader(val height: Int, val blockHeader: BlockHeader) : ElectrumClientAction
-    data class SendResponse(val response: ElectrumResponse) : ElectrumClientAction
+    data class SendResponse(val ids: Set<UUID>, val response: ElectrumResponse) : ElectrumClientAction
```

Comment:
what is the use case for having multiple different ids here ?

Reply:
It started as a way to handle subscriptions, but I abandonned that because it seemed like too much work. Instead I went for a broadcast behavior, but I guess a nullable id would work. Let me try that.

Reply:
Good catch, done in 13ca0c4. Also use a simpler id in 1ca2ba4.